lib/primitives/covenant.js

```diff
@@ -334,6 +341,232 @@ class Covenant extends bio.Struct {
+    this.items = [];
+    return this;
+  }
+
+  /**
+   * Set covenant to CLAIM.
+   * @param {Hash} nameHash
+   * @param {Buffer} rawName
+   * @returns {Covenant}
+   */
+
+  setOpen(nameHash, rawName) {
+    this.type = types.OPEN;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(0);
+    this.push(rawName);
+    return this;
+  }
+
+  /**
+   * Set covenant to BID.
+   * @param {Hash} nameHash
+   * @param {Number} start
+   * @param {Buffer} rawName
+   * @param {Hash} blind
+   * @returns {Covenant}
+   */
+
+  setBid(nameHash, start, rawName, blind) {
+    this.type = types.BID;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(start);
+    this.push(rawName);
+    this.pushHash(blind);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to REVEAL.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @param {Hash} nonce
+   * @returns {Covenant}
+   */
+
+  setReveal(nameHash, height, nonce) {
+    this.type = types.REVEAL;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+    this.pushHash(nonce);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to REDEEM.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @returns {Covenant}
+   */
+
+  setRedeem(nameHash, height) {
+    this.type = types.REDEEM;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to REGISTER.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @param {Buffer} record
+   * @param {Hash} blockHash
+   * @returns {Covenant}
+   */
+
+  setRegister(nameHash, height, record, blockHash) {
+    this.type = types.REGISTER;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+    this.push(record);
+    this.pushHash(blockHash);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to UPDATE.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @param {Buffer} resource
+   * @returns {Covenant}
+   */
+
+  setUpdate(nameHash, height, resource) {
+    this.type = types.UPDATE;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+    this.push(resource);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to RENEW.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @param {Hash} blockHash
+   * @returns {Covenant}
+   */
+
+  setRenew(nameHash, height, blockHash) {
+    this.type = types.RENEW;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+    this.pushHash(blockHash);
+
+    return this;
+  }
+
+  /**
+   * Set covenant to TRANSFER.
+   * @param {Hash} nameHash
+   * @param {Number} height
+   * @param {Address} address
+   * @returns {Covenant}
+   */
+
+  setTransfer(nameHash, height, address) {
+    this.type = types.TRANSFER;
+    this.items = [];
+    this.pushHash(nameHash);
+    this.pushU32(height);
+    this.pushU8(address.version);
+    this.push(address.hash);
```

Comment:
Would an `assert(address instanceof Address)` make sense here? pushU8 and push will catch them anyway ig.